src/traces/heatmap/calc.js

```diff
@@ -57,10 +57,15 @@ module.exports = function calc(gd, trace) {
         z = binned.z;
     }
     else {
-        if(hasColumns(trace)) convertColumnData(trace, xa, ya, 'x', 'y', ['z']);
+        if(hasColumns(trace)) {
+            convertColumnData(trace, xa, ya, 'x', 'y', ['z']);
+            x = trace.x;
+            y = trace.y;
```

Comment:
> were we double-converting triplet

Right. Previously `ax.d2c` was called twice in `heatmap/calc.js`. For example, on `master` the [`heatmap_xyz-dates-and-categories`](http://rickyreusser.com/plotly-mock-viewer/#heatmap_xyz-dates-and-categories) mock has:

```js
gd._fullLayout.yaxis._categories
// ["3-month", "6-month", "1-year", "2-year", "3-year", "5-year", 0, 1, 2, 3, 4, 5]
```

which still gives the correct result though as the string categories are ignored during `r2p`. 

Now that, starting in this PR, `r2p` convert strings to pixels, this patch is now necessary to make the [`heatmap_xyz-dates-and-categories`](https://github.com/plotly/plotly.js/blob/master/test/image/baselines/heatmap_xyz-dates-and-categories.png) pass.

> Do we have a test of these cases?

Good call. I added `heatmap/calc` test case to :lock: this down more explicitly in 4434fadf634c3e0a877706fea38dc9358bce9d09